### ISSUE TITLE
Running `pytest` in an environment with `nameko` installed screws up interactive user input for remainder of the shell session

### ISSUE THREAD

**Issue by DomKennedy** [nameko/nameko#667]
This problem has been plaguing me for months, but I have only just now discovered that it is being caused by Nameko.

**Describe the bug**

When running a Linux terminal command which is expected to "pause" interactively for user input (for example `git add -p`, or, for a minimal example, `pip uninstall`, which displays a yes/no prompt before continuing), the "pausing" behaviour does not work correctly - instead, the command will error out when it hits the user input point.

When this problem occurs, it persists for the duration of a shell session, requiring a new terminal window to be opened to make these commands work correctly again.

For a concrete example, this is the output of running `pip uninstall nameko` while this problem is in effect:

```
Uninstalling nameko-2.12.0:
  Would remove:
    ~/src/sandbox/.venv/bin/nameko
    ~/src/sandbox/.venv/lib/python3.6/site-packages/nameko-2.12.0.dist-info/*
    ~/src/sandbox/.venv/lib/python3.6/site-packages/nameko/*
Proceed (y/n)? ERROR: Exception:
Traceback (most recent call last):
  File "~/src/sandbox/.venv/lib/python3.6/site-packages/pip/_internal/cli/base_command.py", line 153, in _main
    status = self.run(options, args)
  File "~/src/sandbox/.venv/lib/python3.6/site-packages/pip/_internal/commands/uninstall.py", line 79, in run
    auto_confirm=options.yes, verbose=self.verbosity > 0,
  File "~/src/sandbox/.venv/lib/python3.6/site-packages/pip/_internal/req/req_install.py", line 755, in uninstall
    uninstalled_pathset.remove(auto_confirm, verbose)
  File "~/src/sandbox/.venv/lib/python3.6/site-packages/pip/_internal/req/req_uninstall.py", line 388, in remove
    if auto_confirm or self._allowed_to_proceed(verbose):
  File "~/src/sandbox/.venv/lib/python3.6/site-packages/pip/_internal/req/req_uninstall.py", line 431, in _allowed_to_proceed
    return ask('Proceed (y/n)? ', ('y', 'n')) == 'y'
  File "~/src/sandbox/.venv/lib/python3.6/site-packages/pip/_internal/utils/misc.py", line 245, in ask
    response = input(message)
EOFError
```

**To Reproduce**

Run the following commands in a shell:
```
$ virtualenv .venv
$ . .venv/bin/activate
$ pip install nameko pytest
$ touch test.py
$ pytest test.py
$ pip uninstall nameko
```
The final command produces the traceback displayed above

**Expected behavior**

Interactive commands work as intended, stopping for user input and continuing once it is provided.

**Environment:**

 - Nameko version: 2.12.0
 - Python version: 3.6.8
 - OS: Ubuntu 18.04.3 LTS

**Comment by jayvdb** [comment]
#703 is probably the underlying problem.

**Comment by mattbennett** [comment]
I can't repro this but @jayvdb is probably right that #703 is the underlying cause. Nameko's pytest plugin is applying the eventlet monkey patch (required for all tests that actually use Nameko) and it is messing up the interactive prompts.

In this case I would guess that it's caused by a bug in eventlet. Which version of eventlet were you using?

**Comment by mattbennett** [comment]
The automatic monkey patching that caused this problem has been fixed in Nameko 2.14.0 and 3.0.0rc10
